contracts/examples/SimpleInterestTermsContract.sol

```diff
@@ -99,20 +112,82 @@ contract SimpleInterestTermsContract is TermsContract {
         view
         returns (uint _expectedRepaymentValue)
     {
-        bytes32 parameters = debtRegistry.getTermsContractParameters(agreementId);
-        uint issuanceBlockTimestamp = debtRegistry.getIssuanceBlockTimestamp(agreementId);
+        SimpleInterestParams memory params = unpackParamsForAgreementID(agreementId);
+
+        if (timestamp <= params.startTimestamp) {
+            /* The query occurs before the contract was even initialized so the
+            expected value of repayments is 0. */
+            return 0;
+        } else if (timestamp >= params.endTimestamp) {
+            /* the query occurs beyond the contract's term, so the expected
+            value of repayment is the full principal plus interest. */
+            return params.principalPlusInterest;
+        } else {
+            uint numUnits = determineNumberOfAmortizationUnitsForTimestamp(timestamp, params);
+            return params.principalPlusInterest.mul(numUnits).div(params.termLengthInAmortizationUnits);
+        }
+    }
+
+    function determineNumberOfAmortizationUnitsForTimestamp(
+        uint timestamp,
+        SimpleInterestParams params
+    )
+        internal
+        returns (uint units)
+    {
+      uint delta = timestamp.sub(params.startTimestamp);
+      if (params.amortizationUnitType == AmortizationUnitType.HOURS) {
+          return delta.div(HOUR_LENGTH_IN_SECONDS);
+      } else if (params.amortizationUnitType == AmortizationUnitType.DAYS) {
+          return delta.div(DAY_LENGTH_IN_SECONDS);
+      } else if (params.amortizationUnitType == AmortizationUnitType.WEEKS) {
+          return delta.div(WEEK_LENGTH_IN_SECONDS);
+      } else if (params.amortizationUnitType == AmortizationUnitType.MONTHS) {
+          // TODO(kayvon): improve this naive implementation. Not all months have 30 days.
+          return delta.div(MONTH_LENGTH_IN_SECONDS);
+      } else if (params.amortizationUnitType == AmortizationUnitType.YEARS) {
+          // TODO(kayvon): improve this naive implementation. Not all years have 365 days.
+          return delta.div(YEAR_LENGTH_IN_SECONDS);
+      }
+    }
+
+    function unpackParamsForAgreementID(
+        bytes32 agreementId
+    )
+        internal
+        returns (SimpleInterestParams params)
+    {
+      bytes32 parameters = debtRegistry.getTermsContractParameters(agreementId);
```

Comment:
Need to add some sort of check for what happens when no debt agreement with `agreementId` exists in the `DebtRegistry`

Reply:
So I used an existing modifier `onlyExtantEntry` in the `DebtRegistry` to ensure that the call to `getTermsContractParameters` throws if the `issuanceHash` is invalid.

I imagine we'll want to add this protection to most other function calls in `DebtRegistry` but considering that out of scope for now. We can add a task to pivotal.